dotnet/UnityBridge/UnityConverters.cs

```diff
@@ -153,15 +156,101 @@ public static void SetAra3DMatrix(this Transform transform, float[] mtx)
 
         public static void SetFromNode(this Transform transform, ISceneNode node)
         {
-            if (!System.Numerics.Matrix4x4.Decompose(node.Transform, out var scl, out var rot, out var pos))
+            // TODO: Strong assumption - The coordinate system of the ISceneNode's Transform matches that of Revit.
+            transform.SetFromRevitMatrix(node.Transform);
+        }
+
+        public static void SetFromRevitSceneNode(this Transform transform, RevitSceneNode node)
+        {
+            transform.SetFromRevitMatrix(node.Transform);
+        }
+
+        public static void SetFromRevitMatrix(this Transform transform, System.Numerics.Matrix4x4 matrix)
+        {
+            if (!matrix.UnityPRS(out var pos, out var rot, out var scl))
                 throw new Exception("Can't decompose matrix");
-            
+
+            RevitToUnityCoords(pos, rot, scl, out var outPos, out var outRot, out var outScl);
+
+            transform.position = outPos;
+            transform.rotation = outRot;
+            transform.localScale = outScl;
+        }
+
+        /// <summary>
+        /// Converts the given Revit-based coordinates into Unity coordinates.
+        /// </summary>
+        public static void RevitToUnityCoords(
```

Comment:
We shouldn't have references to Revit specifically in Ara3D.

This could be renamed to something like "Ara3DToUnity" or something similar

Reply:
Also, it probably shouldn't be "To Unity".  Instead "YupRH" coordinate system (or whatever Unity is)

Reply:
Renamed "ToUnityCoords"